src/pretix/plugins/sendmail/views.py

```diff
@@ -126,6 +175,170 @@ def form_invalid(self, form):
+                    for k, v in get_available_placeholders(self.request.event, self.context_parameters).items():
+                        context_dict[k] = '<span class="placeholder" title="{}">{}</span>'.format(
+                            _('This value will be replaced based on dynamic parameters.'),
+                            v.render_sample(self.request.event)
+                        )
+
+                    subject = bleach.clean(form.cleaned_data['subject'].localize(l), tags=[])
+                    preview_subject = format_map(subject, context_dict)
+                    message = form.cleaned_data['message'].localize(l)
+                    preview_text = markdown_compile_email(format_map(message, context_dict))
+
+                    self.output[l] = {
+                        'subject': _('Subject: {subject}').format(subject=preview_subject),
+                        'html': preview_text,
+                        'attachment': form.cleaned_data.get('attachment')
+                    }
+
+            self.object_count = ocnt
+            return self.get(self.request, *self.args, **self.kwargs)
+
+        self.task.apply_async(
+            kwargs=self.get_task_kwargs(form, objects)
+        )
+        self.request.event.log_action(
+            self.ACTION_TYPE,
+            user=self.request.user,
+            data=dict(form.cleaned_data)
+        )
+        messages.success(self.request, _('Your message has been queued and will be sent to the contact addresses of %s '
+                                         'in the next few minutes.') % self.describe_match_size(len(objects)))
+
+        return redirect(self.get_success_url())
+
+    def get_context_data(self, *args, **kwargs):
+        ctx = super().get_context_data(*args, **kwargs)
+        ctx['output'] = getattr(self, 'output', None)
+        ctx['match_size'] = self.describe_match_size(getattr(self, 'object_count', None))
+        ctx['form_fragment_name'] = self.form_fragment_name
+        ctx['is_preview'] = self.request.method == 'POST' and self.request.POST.get('action') == 'preview' and ctx['form'].is_valid()
+        ctx['view_title'] = self.TITLE
+        return ctx
+
+    def get_form(self, form_class=None):
+        f = super().get_form(form_class)
+        if self.request.method == 'POST' and self.request.POST.get('action') == 'preview':
+            if f.is_valid():
+                for fname, field in f.fields.items():
+                    field.widget.attrs['disabled'] = 'disabled'
+        return f
+
+
+class OrderSendView(BaseSenderView):
+    form_class = forms.OrderMailForm
+    form_fragment_name = "pretixplugins/sendmail/send_form_fragment_orders.html"
+    context_parameters = ['event', 'order', 'position_or_address']
+    task = send_mails_to_orders
+
+    ACTION_TYPE = 'pretix.plugins.sendmail.sent'
+    TITLE = _("Orders or attendees")
+    DESCRIPTION = _("Send an email to every customer who purchased a ticket, to every person a ticket has been "
```

Comment:
I think we need to be careful with the word „ticket“ here. AFAICT emails to customers are sent to any customer, even ones who only bought non-admission products. The second part „every person a ticket has been purchased for“ still stands currently, but if we decide to change the limitation that only admission products can be personalized, then the word „ticket“ does not work in this context as well. Maybe just shorten it?

```suggestion
    DESCRIPTION = _("Send an email to every customer, or to every person a ticket has been "
```